app/controllers/talks_controller.rb

```diff
@@ -38,7 +38,18 @@ def destroy
 
   def upvote
     @talk = Talk.find(params[:id])
-    @talk.votes.create
+    @newvote = true
+
+    # Determines whether or not a user has upvoted the current topic
+    @talk.votes.each do |x|
+      if x.user_id == current_user.id
+        @newvote = false
+      end
+    end
+
+    if @newvote
+      @talk.votes.create(:user_id => current_user.id)
```

Comment:
Please use Ruby 2.0+ style guides (hash rockets have been deprecated). Let's keep the code uniformed so please change to the current style.
`@talk.votes.create(user_id: current_user.id)`